synapse/daemon.py

```diff
@@ -236,7 +241,8 @@ def _loadDmonConf(self, conf):
 
             self.locs[name] = item
             if isinstance(item, EventBus):
-                self.onfini(item.fini)
+                # Insert ebus items in LIFO order for fini
+                self._fini_items.insert(0, item)
```

Comment:
It's a nit, but list.insert is *really* inefficient.  Seems like maybe we should just throw a reverse() around the iter in the fini() callback :)

Reply:
( obvs not a perf issue here, but a pattern i'd like to try to avoid if at all possible )

Reply:
this is fixed